File: ForwardList/Element.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ForwardList
{
	internal class Element<T>
	{
		
		
		
		public T Data { get; set; }		//значение элемента
		public Element<T> pNext { get; set; }  //Следующий элемент
		public Element(T data, Element<T> pNext = null)
		{
			this.Data = data;
			this.pNext = pNext;
            Console.WriteLine($"EConstructor:\t{GetHashCode()}");
        }

		~Element() 
		{ 
            Console.WriteLine($"EDestructor:\t{GetHashCode()}");
		}

	}
}

File: ListTwo/Element.cpp
#include "Element.h"
#include <iostream>
using std::cout;
using std::endl;

template<typename T>
Element<T>::Element(T Data, Element* pNext, Element* pPrev)
	: Data(Data), pNext(pNext), pPrev(pPrev)
{
	cout << "Econstructor:\t" << this << endl;
}

template<typename T>
Element<T>::~Element()
{
	cout << "EDestructor:\t" << this << endl;
}
template class Element<int>;